src/vaults/VaultManager.ts

```diff
@@ -806,7 +806,8 @@ class VaultManager {
     tran?: DBTransaction,
   ): AsyncGenerator<Buffer, void, void> {
     if (tran == null) {
-      const handleInfoRequest = (tran) => this.handleInfoRequest(vaultId, tran);
+      const handleInfoRequest = (tran: DBTransaction) =>
+        this.handleInfoRequest(vaultId, tran);
       return yield* this.db.withTransactionG(async function* (tran) {
         return yield* handleInfoRequest(tran);
       });
```

Comment:
This seems rather pointless, it's used once. Maybe just inline it?

Reply:
Also later it says `const efs = this.efs;` also seems pointless, make the code simpler... no need for internal assignment.

Reply:
Generic generators don't handle the `this` context very well compared to a normal arrow function. Anything that makes use of `this` in a generator needs to be put into a variable or an arrow function outside the generator for it to work properly. We do it in a bunch of places.

Reply:
Don't do the binding pattern instead?

Reply:
As a personal rule I'll only use `.bind` if I absolutely have to. In this case I think it's cleaner to still treat it as a closure but to bind what we need to a variable just before it. it *looks* unnecessary, but its still very clear what's happening.

Reply:
I think `const efs = this.efs;` is however useless though right?